tikv/tikv
# br: implement/refactor a generic interface for snapshot backup and restore

### What is changed and how it works?
Backup
Provide an interface for check admin operator

Restore
Provide 2 interfaces for ebs-based snapshot restore
after the ebs-based restore start, it first assigns a leader, then waits for a leader to apply to the last index, after it, it also waits to apply follower to apply to the last index.
How to use these interfaces, please check test file

once a leader is assigned, it need to wait leader apply to last firstly and then broadcast all region to wait apply to last.

Issue Number: Close #13376

What's Changed:

```
implement/refactor 3 generic interfaces for EBS snapshot backup and restore
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
some mannual test made:
start tidb cluster via tidb-operator
backup tidb cluster
restore tidb cluster while it calls RecoveryLeaderWaitApply to wait leader apply to the last index, and then call RecoveryFollowerWaitApply
check restore and run admin checksum
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Please add a release note.

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with None.
If this PR will be picked to release branch, then a release note is probably required.
```
